### DISCUSSION TITLE
Windows Temperature Sensors

### DISCUSSION THREAD

**Discussion by Renkin42** [timmo001/system-bridge#635] [Ideas] [answered]
### Description

I was hoping I'd be able to replace Open Hardware Monitor with this for monitoring through Home Assistant since I really only need the basics and this seems much sleeker, however it appears the API isn't able to grab my CPU temperature, it just reports as Null. Additionally the CPU voltage is blank, though I'm not especially concerned about that. I know some API's have had issues with Ryzen processors in the past. Is that what's going on here?

### What Platform / OS are you running?

Windows

### What version are you running?

2.0.0

### Anything in the logs or a references that might be useful?

```txt
{
  "cache": {
    "l1d": 0,
    "l1i": 0,
    "l2": 3145728,
    "l3": 16777216
  },
  "cpu": {
    "manufacturer": "AMD",
    "brand": "Ryzen 5 2600 Six-Core Processor",
    "vendor": "AuthenticAMD",
    "family": "23",
    "model": "8",
    "stepping": "2",
    "revision": "2050",
    "voltage": "",
    "speed": 3.4,
    "speedMin": 3.4,
    "speedMax": 3.4,
    "governor": "",
    "cores": 12,
    "physicalCores": 6,
    "processors": 1,
    "socket": "AM4",
    "flags": "de pse msr sep mtrr mca cmov psn clfsh ds mmx fxsr sse sse2 ss htt tm ia64 pbe",
    "virtualization": false,
    "cache": {
      "l1d": 0,
      "l1i": 0,
      "l2": 3145728,
      "l3": 16777216
    }
  },
  "currentSpeed": {
    "min": 3.4,
    "max": 3.4,
    "avg": 3.4,
    "cores": [
      3.4,
      3.4,
      3.4,
      3.4,
      3.4,
      3.4,
      3.4,
      3.4,
      3.4,
      3.4,
      3.4,
      3.4
    ]
  },
  "temperature": {
    "main": null,
    "cores": [],
    "max": null,
    "socket": [],
    "chipset": null
  }
}
```

**Comment by timmo001** [comment]
I have the same issue. According to the [System Information](https://systeminformation.io/cpu.html) docs the WMIC module can have issues unless run as administrator. I have tried this, but it still doesn't report these sensors.

_I don't think this app will ever be a replacement for OHM or other monitoring tools, but I think thermal sensors would be a nice addition if they work cross platform_

**Reply by timmo001** [reply]
This appears to be an issue with WMIC which has issues. An alternative solution might require using another library that implements its own sensors. Browsing the source of OHM may help with this

https://github.com/openhardwaremonitor/openhardwaremonitor

**Reply by timmo001** [reply]
Found/Created a library based on OHM, which should allow us to access this data using a .NET app.

https://github.com/timmo001/system-bridge-windows-sensors

**Comment by FaserF** [comment]
The interesting thing is, that it worked for me on V2.0.0 with Windows, but stopped working with V2.1.0
I am still having the temperature entity in HA.

**Comment by timmo001** [comment, marked as answer]
Windows Temperatures can now be seen using the Libre Hardware Monitor Library which is built into the CPU and Graphics request. These update the other values if different to what is reported by System Information / Node.

_This may require System Bridge to be ran as administrator_
